### ISSUE TITLE
An error occurred deploying CRNN using libtorch

### ISSUE THREAD

**Issue by CallHoganZhang** [pytorch/pytorch#42350]
## To Reproduce

I trained the model on Pytorch and deployed it on libtorch, which I tried in both versions 1.1 and 1.4. They all run successfully in CPU mode, but in GPU mode, the model can be loaded into CUDA normally, but once the reasoning is done, the GPU footprint goes up to 100% and the program exits. The error message is:Warning: Weights are not part of single self-help Chunk of memory. This means they need to be compacted at every call, To compact weights again call flatten_parameters(). (_cudnn_impl at.. \.. \ aten \ SRC \ aten \ native \ cudnn \ RNN CPP: 1266)https://github.com/pytorch/pytorch/issues/19053 I tried, but I don't work.And the model architecture of CRNN is the same as in the above questions, so is the program for PTH conversion to PT.

In addition, I use whether I set model.train() or model.eval ()，the result I use model.is_training() is 1, I do not kone this 1 is meanning is training or not .But the output result I use model.is_training is right? 

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by mruberry** [comment]
Sorry to hear you're experiencing this issue, @CallHoganZhang. Do you have a short Python program that can reproduce this issue? Or does it only appear when loading with libtorch?

**Comment by CallHoganZhang** [comment]
Thanks for your reply.Yes,I met this problem just in libtorch and occur in when running cuda model.

**Comment by CallHoganZhang** [comment]
Thanks for your reply.Yes,I met this problem just in libtorch and occur in when running cuda model.``

**Comment by CallHoganZhang** [comment]
Thanks for your reply.Yes,I met this problem just in libtorch and occur in when running cuda model.
`import torch.nn as nn
import torch


class BidirectionalLSTM(nn.Module):
    # Inputs hidden units Out
    def __init__(self, nIn, nHidden, nOut):
        super(BidirectionalLSTM, self).__init__()

        self.rnn = nn.LSTM(nIn, nHidden, bidirectional=True)
        self.embedding = nn.Linear(nHidden * 2, nOut)

    def forward(self, input):

        recurrent, _ = self.rnn(input)
        T, b, h = recurrent.size()
        t_rec = recurrent.view(T * b, h)

        output = self.embedding(t_rec)  # [T * b, nOut]
        output = output.view(T, b, -1)

        return output


class CRNN(nn.Module):
    #                   32    1   37     256  
    def __init__(self, imgH, nc, nclass, nh, n_rnn=2, leakyRelu=False):
        super(CRNN, self).__init__()
        assert imgH % 16 == 0, 'imgH has to be a multiple of 16'

        ks = [3, 3, 3, 3, 3, 3, 2]
        ps = [1, 1, 1, 1, 1, 1, 0]
        ss = [1, 1, 1, 1, 1, 1, 1]
        nm = [64, 128, 256, 256, 512, 512, 512]

        cnn = nn.Sequential()

        def convRelu(i, batchNormalization=False):
            nIn = nc if i == 0 else nm[i - 1]
            nOut = nm[i]
            cnn.add_module('conv{0}'.format(i),
                           nn.Conv2d(in_channels=nIn, out_channels=nOut, kernel_size=ks[i], stride=ss[i], padding=ps[i])
                           )    # padding=(kernel_size-1)/2 if stride=1
            if batchNormalization:
                cnn.add_module('batchnorm{0}'.format(i), nn.BatchNorm2d(nOut))
            if leakyRelu:
                cnn.add_module('relu{0}'.format(i),
                               nn.LeakyReLU(0.2, inplace=True))
            else:
                cnn.add_module('relu{0}'.format(i), nn.ReLU(True))

        convRelu(0)
        cnn.add_module('pooling{0}'.format(0), nn.MaxPool2d(2, 2))  # 64x16x64
        convRelu(1)
        cnn.add_module('pooling{0}'.format(1), nn.MaxPool2d(2, 2))  # 128x8x32
        convRelu(2, True)
        convRelu(3)

        cnn.add_module('pooling{0}'.format(2),
                       nn.MaxPool2d(kernel_size=(2,2),stride=(2,1),padding=(0,1)))  # 256x4x16
        convRelu(4, True)
        convRelu(5)
        cnn.add_module('pooling{0}'.format(3),
                       nn.MaxPool2d((2, 2), (2, 1), (0, 1)))  # 512x2x16
        convRelu(6, True)  # 512x1x16

        self.cnn = cnn
        self.rnn = nn.Sequential(
            BidirectionalLSTM(512, nh, nh),
            BidirectionalLSTM(nh, nh, nclass))

    def forward(self, input):
        # conv features
        # print('---forward propagation---')
        conv = self.cnn(input)
        # print(conv.size()) batch_size*512*1*width
        b, c, h, w = conv.size()
        assert h == 1, "the height of conv must be 1"
        conv = conv.squeeze(2)          # b *512 * width
        conv = conv.permute(2, 0, 1)    # [w, b, c]
        # print(conv.size()) # width batch_size channel
        # rnn features

        output = self.rnn(conv)
        # print(output.size(0))
        # print(output.size())# width*batch_size*nclass
        return output`

This is my code to construct model,and then I will transform the pth model to pt model by torch.jit.trace funciton,just like this
`if __name__ == '__main__':
    model = CRNN(32, 1, 6736, 256)
    model.load_state_dict(torch.load("ocr.pth"))
    model.eval()
    model.to('cuda')
    inputs = torch.ones(1, 1, 32, 160).cuda()
    traced = torch.jit.trace(model, inputs)
    traced.save('ocr.pt')`

But when I am inferring in libtorch in cpu status,I will not met the issue.It can run model safely.When I run this code using cuda model in libtorch,the problem occurs.
`output = module->forward(inputs).toTensor()`
Besides, In libtorch, I can load model in cuda and load tensor in cuda without problem.once I am inferring,the gpu memory occupies 100% and the exe programme is break down.
So,this my problem and looking forward to your answers.

**Comment by mruberry** [comment]
I see, thank you for the update @CallHoganZhang! I've relabeled the issue so the right people can review it.

**Comment by CallHoganZhang** [comment]
Hi, Thanks for your help! I'm still paying attention to my recent issue。
The environment I use now is:
Win10,

VS2017

pytorch1.6

Libtorch1.6 (now using debug version)
======================================================
This is my transformation model program:

import torch
from torchvision import models

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
print(device)

model = models.resnet18()

example = torch.rand(1, 3, 224, 224)
model = model.eval()

traced_script_module = torch.jit.trace(model, example)
output = traced_script_module(torch.ones(1, 3, 224, 224))
traced_script_module.save("model.pt")

And I loaded model in libtorch:

#include <torch/script.h>
#include <iostream>

using namespace std;

int main()
{
	string model_path = "E:/pro/temp/model.pt";
	torch::jit::script::Module module;
	
	module = torch::jit::load(model_path);
	module.to(at::kCUDA);
	cout << "loading model OK" << "\n";

	return 0;
}

So When I using the debug version of libtorch I can normally print the tensor values and just like this:

auto t1 = torch::tensor({ 1,2,3,4,5,6,7,8,9 }).reshape({ 3,3 });
auto t2 = torch::tensor({ 1,0,2,6,1,1,5,3,2 }).reshape({ 3,3 });
auto t3 = t1.mul(t2);
std::cout << t3 << std::endl;


But I can't load the model. And when I use libtorch, 

there's a c++ problem(E1866): new feature that doesn't apply to entities.

Looking forward to your reply.Thanks!

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
